packages/babel-plugin-remove-undefined-if-possible/src/index.js

```diff
@@ -0,0 +1,82 @@
+"use strict";
+
+function removeRvalIfUndefined(declaratorPath) {
+  const rval = declaratorPath.get("init")
+                              .evaluate();
+  if (rval.confident === true && rval.value === undefined) {
```

Comment:
This also needs to be pure.

```js
var x = void console.log("hi");
var y = void foo();
```

shouldn't be evaluated to undefined and be removed.